pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -150,6 +150,14 @@ def advance(  # type: ignore[override]
             # if fault tolerant is enabled and process has been notified, exit.
             self.trainer._exit_gracefully_on_signal()
 
+    def on_advance_start(self) -> None:  # type: ignore[override]
+        if self.trainer.val_dataloader is not None and callable(
```

Comment:
You will need to select the dataloader based on `if self.trainer.testing`

Reply:
Used `self.dataloaders` instead.